ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -135,7 +116,53 @@
 public class KsqlResource {
   // CHECKSTYLE_RULES.ON: ClassDataAbstractionCoupling
 
-  private static final org.slf4j.Logger log = LoggerFactory.getLogger(KsqlResource.class);
+  private static final Map<Class<? extends Statement>, Validator<Statement>> CUSTOM_VALIDATORS =
+      ImmutableMap.<Class<? extends Statement>, Validator<Statement>>builder()
+          .put(Query.class,
+              castValidator(KsqlResource::validateQueryEndpointStatements, Query.class))
+          .put(PrintTopic.class,
+              castValidator(KsqlResource::validateQueryEndpointStatements, PrintTopic.class))
+          .put(SetProperty.class,
+              castValidator(KsqlResource::validatePropertyStatements, SetProperty.class))
+          .put(UnsetProperty.class,
+              castValidator(KsqlResource::validatePropertyStatements, UnsetProperty.class))
+          .put(TerminateQuery.class,
+              castValidator(KsqlResource::skipValidate, TerminateQuery.class))
```

Comment:
The old code didn't validate terminate calls.  Without this custom no-op validator `validateStatements` will validate this call in the same way it would validate any other executable statement.  So this is needed to maintain the existing functionality.